File: 274.h-指数.py
#
# @lc app=leetcode.cn id=274 lang=python3
#
# [274] H 指数
#

# @lc code=start
class Solution:
    def hIndex(self, citations: List[int]) -> int:
        citations.sort()
        cnt, l = 0, len(citations)
        while cnt < l and citations[l - cnt - 1] > cnt:
            cnt += 1
        return cnt
# @lc code=end

File: 274.h-指数.java
import java.util.Arrays;

/*
 * @lc app=leetcode.cn id=274 lang=java
 *
 * [274] H 指数
 */

// @lc code=start
class Solution {
    public int hIndex(int[] citations) {
        Arrays.sort(citations);
        int n = citations.length, cnt = 0;
        while (cnt < n && citations[n - cnt - 1] > cnt)  ++cnt;
        return cnt;
    }
}
// @lc code=end

